src/main/kotlin/com/cosmotech/api/security/SecurityCoroutineContext.kt

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+package com.cosmotech.api.security
+
+import kotlin.coroutines.CoroutineContext
+import kotlinx.coroutines.ThreadContextElement
+import org.springframework.security.core.context.SecurityContext
+import org.springframework.security.core.context.SecurityContextHolder
+
+class SecurityCoroutineContext(
+    private val securityContext: SecurityContext = SecurityContextHolder.getContext()
+) : ThreadContextElement<SecurityContext?> {
+
+  companion object Key : CoroutineContext.Key<SecurityCoroutineContext>
+
+  override val key: CoroutineContext.Key<SecurityCoroutineContext>
```

Comment:
Could you please remove this and test again?
Apparently, it has no effect for the moment, so there's no need to keep it ;)

Reply:
As discuss keep the class for now